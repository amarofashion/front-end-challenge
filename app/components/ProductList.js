import React, { Component, PropTypes } from 'react'
import catalogStl from '../styles/catalog.styl'
import modalStl from '../styles/modal.styl'
import ProductListItem from './product/ProductListItem'

export default class ProductList extends Component {
	static propTypes = {
		products: PropTypes.array
	}

	render() {
		const { products, addToCart } = this.props

		return <section className="catalog">
			<div className="catalog__filter">
				<h3>filter goes here</h3>
			</div>
			<ul className="catalog__list">
				{products.map((product, i) => 
					<ProductListItem key={ `prdct ${i}` } product={ product } addToCart={ addToCart } id={ i }/>
				)}
			</ul>
		</section>
	}
}