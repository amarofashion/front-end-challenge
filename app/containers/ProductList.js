import React, { Component } from 'react'

export default class ProductList extends Component {
	render() {
		const { store, history } = this.props

		return <div>
			Hey! Thanks for using this example. If you like it, consider starring the repo :))
			<div>
				<a href="/product">product</a>
			</div>
		</div>
	}
}