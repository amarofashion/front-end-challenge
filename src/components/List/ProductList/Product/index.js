import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Popup from "reactjs-popup";
import uuid from "uuid";

import Thumb from "../../../Thumb";

import { addProduct } from "../../../../api/redux/reducers/cart/actions";

const Product = ({ product, addProduct }) => {
  product.quantity = 1;

  const productInstallment = (
    <div className="installment">
      <span>ou {product.installments}</span>
    </div>
  );

  return (
    <div className="product-item" data-sku={product.sku}>
      {product.on_sale && (
        <div className="product-promotion">
          {product.discount_percentage && (
            <small>{product.discount_percentage} off</small>
          )}
        </div>
      )}
      <Thumb
        classes="product-item__thumb"
        src={product.image}
        alt={product.title}
      />
      <p className="product-item__title">{product.name}</p>

      <div className="product-item__price">
        <div className="val">
          {product.discount_percentage && (
            <span>De: {product.regular_price} Por:</span>
          )}
          <b>{product.actual_price}</b>
        </div>
        {productInstallment}
      </div>
      <Popup
        modal
        trigger={<button className="product-item__buy-btn">Adicionar</button>}
      >
        {close => (
          <div>
            Por favor selecione uma opção:
            <div className="product-item__size" onClick={close}>
              {product.sizes.map(value => {
                return (
                  <Fragment key={uuid.v4()}>
                    <button
                      disabled={!value.available}
                      onClick={() =>
                        addProduct({
                          ...product,
                          id: value.sku,
                          variant: value.size
                        })
                      }
                    >
                      {value.size}
                    </button>
                  </Fragment>
                );
              })}
            </div>
          </div>
        )}
      </Popup>
    </div>
  );
};

Product.propTypes = {
  product: PropTypes.object.isRequired,
  addProduct: PropTypes.func.isRequired
};

export default connect(
  null,
  { addProduct }
)(Product);
