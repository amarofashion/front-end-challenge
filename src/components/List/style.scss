.product-container {
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    min-height: 600px;

    .product-item {
      width: 23%;
      position: relative;
      text-align: center;
      box-sizing: border-box;
      padding: 10px;
      margin-bottom: 30px;
      border: 1px solid transparent;
      border: 1px solid #eee;
      margin: 10px;
      &:hover {
        .product-item__buy-btn {
          background-color: #000000;
          color: #eee;
        }
      }
  
      .product-promotion {
        position: absolute;
        color: #000;
        border: solid 1px #000;
        top: 10px;
        right: 10px;
        padding: 5px;
        border-radius: 50%;
        font-size: 1em;
        background-color: #fff;
        cursor: default;
      }
  
      &__thumb {
        img {
          width: 100%;
        }
      }
  
      &__title {
        position: relative;
        padding: 0 20px;
        height: 35px;
        @media only screen and (max-width: 1024px) {
           height: 50px;  
        }
      }
      &__size {
          button {
            align-items: center;
            border: .1rem solid #ccc;
            border-radius: 50%;
            display: inline-flex;
            font-size: 1.4rem;
            height: 4.8rem;
            justify-content: center;
            line-height: 1;
            margin-left: .5rem;
            margin-right: .5rem;
            margin-top: 1rem;
            text-decoration: none;
            width: 4.8rem;
            &:disabled {
                background-color: #e6e6e6;
                pointer-events: none;
            }
          }
      }
  
      &__price {
        height: 60px;
        .val {
          b {
            font-size: 1.5em;
            margin-left: 5px;
          }
        }
        .installment {
          color: #000000;
        }
      }
  
      &__buy-btn {
        color: #fff;
        padding: 15px 0;
        margin-top: 10px;
        background-color: Transparent;
        background-repeat: no-repeat;
        cursor: pointer;
        overflow: hidden;
        outline: none;
        width: 100%;
        text-align: center;
        font-weight: bold;
        font-size: 18px;
        text-transform: uppercase;
        color: #000000;
        transition: background-color 0.2s;
      }
    }
  }
  