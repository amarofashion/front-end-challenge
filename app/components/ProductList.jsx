import React, { Component } from 'react';

import Product from './Product.jsx';

class ProductList extends Component {
    render() {
        var {products} = this.props;        

        var renderProducts = () => {
            return products.map((product, index) => {
                return (
                    <div className="col-xs-6">
                        <Product key={product.id} {...product} />
                    </div>
                )
            })
        }

        return (
            <div>
                <div className="row">
                    {renderProducts()}
                </div>
            </div>
        );
    }
}

export default ProductList;