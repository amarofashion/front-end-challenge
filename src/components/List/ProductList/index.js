import React from 'react';
import uuid from "uuid";
import Product from './Product';

const ProductList = ({ products}) => {
  return products.map(p => {
    return <Product product={p} key={uuid.v4()} />;
  });
};

export default ProductList;
