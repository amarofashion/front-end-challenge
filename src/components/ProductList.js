import React from 'react'
import Product from './Product'

const { array, func, string } = React.PropTypes

const ProductList = ({ products, defaultImage, onAddedToCart }) => (
  <div className="row">
    { products.map( p =>
        <Product
          key={ p.id }
          defaultImage={ defaultImage }
          { ...p }
          onAddToCart={onAddedToCart}
        /> ) }
  </div>
)

ProductList.PropTypes = {
  products: array.isRequired,
  defaultImage: string.isRequired,
  onAddedToCart: func
}

export default ProductList