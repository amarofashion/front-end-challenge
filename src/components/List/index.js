import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { getProducts } from '../../api/redux/reducers/list/actions';

import Spinner from '../Spinner';
import ProductList from './ProductList';

import './style.scss';

class List extends Component {
  static propTypes = {
    getProducts: PropTypes.func.isRequired,
    products: PropTypes.array.isRequired,
  };

  state = {
    isLoading: false
  };

  componentDidMount() {
    this.getProducts();
  }

  getProducts = () => {
    this.setState({ isLoading: true });
    this.props.getProducts(() => {
      this.setState({ isLoading: false });
    });
  };

  render() {
    const { products } = this.props;
    const { isLoading } = this.state;

    return (
      <Fragment>
        {isLoading && <Spinner />}
        <div className="product-container">
          <ProductList products={products}/>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  products: state.list.products
});

export default connect(
  mapStateToProps,
  { getProducts }
)(List);
