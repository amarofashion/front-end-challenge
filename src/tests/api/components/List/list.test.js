import React from 'react';
import List from '../../../../components/List/index';
import ProductList from '../../../../components/List/ProductList/index';
import renderer from 'react-test-renderer';
import CustomProvider from '../../../../../src/provider/index';

const dataProduct = [{
    
    name: "CASACO WHITE FUR",
    style: "20002648",
    sku: "5891_1000058_0_PP",
    code_color: "20002648_029",
    color_slug: "off-white",
    color: "OFF-WHITE",
    on_sale: false,
    regular_price: "R$ 239,90",
    actual_price: "R$ 239,90",
    variant: 'PP',
    discount_percentage: "",
    installments: "3x R$ 79,97",
    image: "https://d3l7rqep7l31az.cloudfront.net/images/products/20002648_029_catalog_1.jpg?1459552552",
}];


describe("List Component", () => {
  test("Wrapper list component test", () => {
    const tree = renderer.create( 
    <CustomProvider>
       <List />
    </CustomProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Wrapper product list component test", () => {
    const tree = renderer.create( 
    <CustomProvider>
       <ProductList products={dataProduct}  />
    </CustomProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});