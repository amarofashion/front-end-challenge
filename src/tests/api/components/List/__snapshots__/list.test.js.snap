// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Wrapper list component test 1`] = `
Array [
  <div
    className="spinner animations"
  >
    <div />
    <div />
    <div />
    <div />
  </div>,
  <div
    className="product-container"
  />,
]
`;

exports[`List Component Wrapper product list component test 1`] = `
<div
  className="product-item"
  data-sku="5891_1000058_0_PP"
>
  <div
    className="product-item__thumb"
  >
    <img
      src="https://d3l7rqep7l31az.cloudfront.net/images/products/20002648_029_catalog_1.jpg?1459552552"
    />
  </div>
  <p
    className="product-item__title"
  >
    CASACO WHITE FUR
  </p>
  <div
    className="product-item__price"
  >
    <div
      className="val"
    >
      
      <b>
        R$ 239,90
      </b>
    </div>
    <div
      className="installment"
    >
      <span>
        ou 
        3x R$ 79,97
      </span>
    </div>
  </div>
  <button
    className="product-item__buy-btn"
    onClick={[Function]}
  >
    Adicionar
  </button>
</div>
`;
